tools/restart_network.sh

```diff
@@ -1,14 +1,14 @@
 #!/bin/sh
 
-function warning {
-    echo "`basename $0`: Warning: ${@:1}" 1>&2;
+warning() {
+    printf "%s: Warning: %s\n" "$(basename $0)" "${@}" 1>&2;
```

Comment:
The effect of `"$@"` is not desired here I believe

```suggestion
    printf "%s: Warning: %s\n" "$(basename $0)" "$*" 1>&2
```